Assets/MixedRealityToolkit.Services/InputSystem/GazeProvider.cs

```diff
@@ -293,76 +252,69 @@ public void RaisePointerUp(MixedRealityInputAction mixedRealityInputAction, Hand
                 isDown = false;
                 currentHandedness = Handedness.None;
                 currentInputSource = null;
-                gazeProvider.InputSystem?.RaisePointerClicked(this, mixedRealityInputAction, 0, handedness, inputSource);
-                gazeProvider.InputSystem?.RaisePointerUp(this, mixedRealityInputAction, handedness, inputSource);
+                InputSystem.RaisePointerClicked(this, mixedRealityInputAction, 0, handedness, inputSource);
+                InputSystem.RaisePointerUp(this, mixedRealityInputAction, handedness, inputSource);
             }
         }
 
         #endregion InternalGazePointer Class
 
-        #region MonoBehaviour Implementation
-
-        private void OnValidate()
-        {
-            if (minHeadVelocityThreshold > maxHeadVelocityThreshold)
-            {
-                Debug.LogWarning("Minimum head velocity threshold should be less than the maximum velocity threshold. Changing now.");
-                minHeadVelocityThreshold = maxHeadVelocityThreshold;
-            }
-        }
-
         /// <inheritdoc />
-        protected override void OnEnable()
+        public override void Enable()
         {
-            base.OnEnable();
-
-            if (!delayInitialization)
+            // The lifetime of the gaze provider is controlled by multiple forces:
+            // The MixedRealityToolkit object itself during instantiation (to be consistent with
+            // how the other focus and raycast providers work, and the input system itself (when
+            // it is enabled/disabled). As a result, certain functions have to guard against the
+            // current enabled/disabled state.
+            if (Enabled)
             {
-                // The first time we call OnEnable we skip this.
-                RaiseSourceDetected();
+                return;
             }
-        }
 
-        /// <inheritdoc />
-        protected override async void Start()
-        {
-            base.Start();
-
-            await EnsureInputSystemValid();
-
-            if (this == null)
+#if UNITY_EDITOR
+            if (profile.MinHeadVelocityThreshold > profile.MaxHeadVelocityThreshold)
             {
-                // We've been destroyed during the await.
-                return;
+                Debug.LogWarning("Minimum head velocity threshold should be less than the maximum velocity threshold. Changing now.");
+                profile.MinHeadVelocityThreshold = profile.MaxHeadVelocityThreshold;
             }
+#endif
 
+            base.Enable();
+            CoreServices.InputSystem.RegisterHandler<IMixedRealityInputHandler>(this);
             GazePointer.BaseCursor?.SetVisibility(true);
+            CoreServices.InputSystem.RaiseSourceDetected(GazeInputSource);
```

Comment:
Hm...why is this also an input source? Any ideas of examples where we are using the gaze input source? Seems like device managers should raise input source detected.